ray-operator/controllers/ray/batchscheduler/yunikorn/yunikorn_task_groups.go

```diff
@@ -0,0 +1,96 @@
+package yunikorn
+
+import (
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	v1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/utils"
+)
+
+// TaskGroups is a list of task Groups recognized as gang Groups
+type TaskGroups struct {
+	Groups []TaskGroup `json:"groups"`
+}
+
+// TaskGroup is the struct for yunikorn to consider a pod belongs to a gang group
```

Comment:
I guess Yunikorn has the schema definition for TaskGroup. Would you mind adding a link to the schema's source of truth in the comment?

Reply:
Sure. Let me add it.